osu.Game/Overlays/Settings/Sections/Graphics/RendererSettings.cs

```diff
@@ -41,6 +52,21 @@ private void load(FrameworkConfigManager config, OsuConfigManager osuConfig)
                     Current = osuConfig.GetBindable<bool>(OsuSetting.ShowFpsDisplay)
                 },
             };
+
+            renderer.BindValueChanged(r =>
+            {
+                if (r.NewValue == host.ResolvedRenderer)
+                    return;
+
+                // Need to check startup renderer for the "automatic" case, as ResolvedRenderer above will track the final resolved renderer instead.
+                if (r.NewValue == RendererType.Automatic && automaticRendererInUse)
+                    return;
+
+                dialogOverlay.Push(new ConfirmDialog(GraphicsSettingsStrings.ChangeRendererConfirmation, game.AttemptExit, () =>
```

Comment:
Does `AttemptExit()` work on iOS? It sure doesn't work on android (doesn't kill the activity, it just stays there on a black screen).

Reply:
As per discord [it doesn't](https://discord.com/channels/188630481301012481/188630652340404224/1089511493227249835), but I'm going to look away for now. Hopefully users can figure it out on their own (and we'll probably fix properly anyway once android support is in, via a splash screen or something).